CMakeLists.txt

```diff
@@ -15,20 +15,24 @@ if("${CMAKE_SYSTEM_NAME}" MATCHES ".*[Ll]inux.*")
     set(LINUX TRUE CACHE BOOL "..." FORCE) # LINUX means GNU/Linux, not just the kernel
 elseif("${CMAKE_SYSTEM_NAME}" MATCHES ".*[Bb][Ss][Dd].*")
     set(BSD TRUE CACHE BOOL "..." FORCE)
-elseif(NOT APPLE AND NOT ANDROID AND NOT MSYS)
+elseif(NOT APPLE AND NOT ANDROID AND NOT MSYS AND NOT WIN32)
     message(FATAL_ERROR "Unsupported platform: ${CMAKE_SYSTEM_NAME}")
 endif()
 
-if(MSYS)
+if(MSYS OR WIN32)
     enable_language(CXX)
 endif()
 
 #############################
 # Compile time dependencies #
 #############################
 
-set(THREADS_PREFER_PTHREAD_FLAG ON)
-find_package(Threads)
+if(WIN32)
+    set(Threads_FOUND ON) # Force using Win32 Threads on Windows, even if winpthread is found
+else()
+    set(THREADS_PREFER_PTHREAD_FLAG ON)
+    find_package(Threads)
+endif()
```

Comment:
Not against the current implementation, but i think it would be cleaner to prefer pthread only on linux. It is recommended there by the [doc](https://cmake.org/cmake/help/latest/module/FindThreads.html#variables-affecting-behavior) but unneeded on other platforms.

```cmake
set(THREADS_PREFER_PTHREAD_FLAG ${NOT WIN32})
find_package(Threads)
```